src/frontend/src/hooks/useExcelImport.ts

```diff
@@ -0,0 +1,107 @@
+import { useState, useCallback, useMemo, useEffect } from 'react';
+import { useAppContext } from '../appContext';
+import ExcelParseReponse, { ExcelHeader } from '../models/ExcelParseResponse';
+
+export type autoGenerateColumn<T> = {
+    title: keyof T;
+    format: (columns: Column<T>[]) => {};
+};
+
+export type Column<T> = {
+    title: keyof T;
+    variations?: string[];
+    format?: (item: string) => {};
+};
+
+export type ExcelImportSettings<T> = {
+    columns: Column<T>[];
+    autoGenerateColumns?: autoGenerateColumn<T>[];
+};
+
+type ColumnIndex<T> = {
+    [key in keyof T]: number;
+};
+
+const useExcelImport = <T>(excelImportSettings: ExcelImportSettings<T>) => {
+    const { apiClient } = useAppContext();
+
+    const [selectedFile, setSelectedFile] = useState<File | null>(null);
+    const [isProccessingFile, setIsProccessingFile] = useState<boolean>(false);
+    const [processedFile, setProcessedFile] = useState<T[] | null>(null);
+    const [error, setError] = useState<boolean>(false);
+    const { columns, autoGenerateColumns } = excelImportSettings;
+
+    useEffect(() => {
+        if (selectedFile === null) return;
+
+        setIsProccessingFile(true);
+        getExcelReponseAsync(selectedFile);
+    }, [selectedFile]);
+
+    const getExcelReponseAsync = async (file: File) => {
+        const excelReponse = await apiClient.ExcelImportParserAsync(file);
```

Comment:
Should be wrapped with try/catch. And `setIsProcessingFile(true);` should probably be in the function instead of the `useEffect` above

Reply:
Reorganised this a little bit. 
Renaming the getExcelResponse function to processExcelFileAsync.
Which in turn now has a try/catch and also handles all state changes. 
Making the underlying functions more pure, and the general flow abit easier to follow.